components/datepicker/DatePicker.js

```diff
@@ -0,0 +1,65 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import DayPicker from 'react-day-picker';
+import NavigationBar from './NavigationBar';
+import WeekDay from './WeekDay';
+import { convertModifiersToClassnames } from './utils';
+import cx from 'classnames';
+import theme from './theme.css';
+
+class DatePicker extends PureComponent {
+  state = {
+    selectedDate: null,
+  };
+
+  componentWillMount() {
```

Comment:
`componentWillMount` is deprecated (16.3); checking out [this article](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#initializing-state), initializing state can be done right in the component without lifecycle method
```jsx
class DatePicker extends PureComponent {
	state = {
		selectedDate: this.props.selectedDate || null,
	};
}
```